### ISSUE TITLE
[s2s] a later added password removes the remote share

### ISSUE THREAD

**Issue by DeepDiver1975** [owncloud/core#13564]

**Comment by PVince81** [comment]
Possibly duplicate of https://github.com/owncloud/core/issues/10671

**Comment by rullzer** [comment]
Could you check if this is fixed as well after #14987

**Comment by PVince81** [comment]
Adding/changing the password of a s2s link share makes it inaccessible for recipients. The current logic is as follows: if the shareinfo.php call returns 403 or 401, it means that the password changed, so the share isn't accessible any more. Currently it will be deleted. In the future we should implement a dialog that re-prompts for the new password of that share for the user to keep it.

I didn't find any ticket to add a password prompt, so let's reuse this one for this enhancement.

**Comment by PVince81** [comment]
This is expected behavior. We need to implement a password prompt. Maybe this can be done somehow as part of the notifications ?

Moving to 9.0 (there is no possible quickfix)

**Comment by PVince81** [comment]
Backlog for now
